FWCore/Concurrency/src/async.cc

```diff
@@ -0,0 +1,30 @@
+#include "FWCore/Concurrency/interface/async.h"
+
+namespace edm::impl {
+  WaitingThread::WaitingThread() { thread_ = std::thread(&WaitingThread::threadLoop, this); }
```

Comment:
Could you set a meaningful thread name, so it's easier to identify this thread pool in a GDB trace ?

For example:
```suggestion
  WaitingThread::WaitingThread() {
    thread_ = std::thread(&WaitingThread::threadLoop, this);
    pthread_setname_np(thread_.native_handle(), "edm async pool");
  }
```

Or even something more elaborate with a `static constexpr` name, and a check that its length is 15 or less.

Reply:
Thanks for the suggestion, I added the name. While verifying the behavior by running the unit test in `gdb` I discovered many threads (~8) with the `edm async pool` name. Further investigation showed that the test ended up creating two `WaitingThread`s (which I can believe), but the use of `global_control` to set allowed parallelism to 1 lead the call to `onetbb::task_arena::enqueue()` in `edm::WaitingTaskWithArenaHolder::doneWaiting()` to create a new TBB-controlled thread that inherited the `edm async pool` name (and all subsequent TBB threads created by that thread inherited the name too).

When I set the allowed parallelism to 2 (which is now done in the test), I saw 2 threads with the `edm async pool` name.

While nearly all production jobs are configured to use multiple threads, maybe it would be time to look more into trying to give names to the TBB threads etc. I'll open a separate issue on that.